services/src/main/java/org/keycloak/authentication/otp/GoogleAuthenticatorProvider.java

```diff
@@ -22,14 +22,6 @@ public String getName() {
 
     @Override
     public boolean supports(OTPPolicy policy) {
-        if (policy.getDigits() != 6) {
-            return false;
-        }
-
-        if (!policy.getAlgorithm().equals("HmacSHA1")) {
-            return false;
-        }
-
         return policy.getType().equals("totp") && policy.getPeriod() == 30;
```

Comment:
I think google authenticator also supports `hotp` now. Can you please test?
If that the case  then the only limitation is that it just supports 30s for totp.

Reply:
Oh, that is true! hotp also worked now!  Updated accordingly.